CS0540 partial type with generic base does not implement interface

**Version Used**: `4.4.0-2.22405.13 (dcfe294d)` / `7.0.100-preview.4.22252.9`

**Steps to Reproduce**:

**Test1.cs**
```C#
interface ITest
{
    void Test();
}

class Generic<T> { }
class Argument { }
partial class Partial : Generic<Argument> { }
```

**Test1.cs**
```C#
partial class Partial : Generic<Argument>, ITest
{
    void ITest.Test() { }
}
```

Now either
1. include `#nullable enable` in Test1.cs; or
2. set `<Nullable>enable</Nullable>` in the project and add `// <auto-generated />` to Test2.cs

**Expected Behavior**: Compiles without errors. As per [Partial and Methods](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods) "Parts can specify different base interfaces, and the final type implements all the interfaces listed by all the partial declarations."

**Actual Behavior**:
```
Test2.cs(5,10,5,15): error CS0540: 'Partial.ITest.Test()': containing type does not implement interface 'ITest'
```

**Related:**
#32041, #31858

--- a/src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs
@@ -324,18 +324,24 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
                             {
                                 baseType = partBase;
                                 baseTypeLocation = decl.NameLocation;
-                                continue;
                             }
-                            else if (containsOnlyOblivious(partBase))
+                            else if (!containsOnlyOblivious(partBase))
                             {
-                                continue;
+                                reportBaseType();
                             }
                         }
+                        else
+                        {
+                            reportBaseType();
+                        }
 
-                        var info = diagnostics.Add(ErrorCode.ERR_PartialMultipleBases, Locations[0], this);
-                        baseType = new ExtendedErrorTypeSymbol(baseType, LookupResultKind.Ambiguous, info);
-                        baseTypeLocation = decl.NameLocation;
-                        reportedPartialConflict = true;
+                        void reportBaseType()
+                        {
+                            var info = diagnostics.Add(ErrorCode.ERR_PartialMultipleBases, Locations[0], this);
+                            baseType = new ExtendedErrorTypeSymbol(baseType, LookupResultKind.Ambiguous, info);
+                            baseTypeLocation = decl.NameLocation;
+                            reportedPartialConflict = true;
+                        }
 
                         static bool containsOnlyOblivious(TypeSymbol type)
                         {

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/InheritanceBindingTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/InheritanceBindingTests.cs
@@ -9794,5 +9794,29 @@ class C2 : I
 ";
             var comp = CreateCompilation(source).VerifyDiagnostics();
         }
+
+        [Fact]
+        [WorkItem(63490, "https://github.com/dotnet/roslyn/issues/63490")]
+        public void MultipleBasesWithObliviousDifferencesAndInterfaces()
+        {
+            var source = @"
+#nullable enable
+interface ITest
+{
+    void Test();
+}
+
+class Generic<T> { }
+class Argument { }
+partial class Partial : Generic<Argument> { }
+
+#nullable disable
+partial class Partial : Generic<Argument>, ITest
+{
+    void ITest.Test() { }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics();
+        }
     }
 }

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/InheritanceBindingTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/InheritanceBindingTests.cs
@@ -9799,7 +9799,7 @@ class C2 : I
         [WorkItem(63490, "https://github.com/dotnet/roslyn/issues/63490")]
         public void MultipleBasesWithObliviousDifferencesAndInterfaces()
         {
-            var source = @"
+            var source1 = @"
 #nullable enable
 interface ITest
 {
@@ -9809,14 +9809,17 @@ interface ITest
 class Generic<T> { }
 class Argument { }
 partial class Partial : Generic<Argument> { }
+";
 
+            var source2 = @"
 #nullable disable
 partial class Partial : Generic<Argument>, ITest
 {
     void ITest.Test() { }
 }
 ";
-            CreateCompilation(source).VerifyDiagnostics();
+            CreateCompilation(source1 + source2).VerifyDiagnostics();
+            CreateCompilation(source2 + source1).VerifyDiagnostics();
         }
     }
 }

